src/createAsset.ts

```diff
@@ -181,10 +88,11 @@ export class CreateAsset {
 		);
 		const posModuleAssets = await addPoSModuleEntry(
 			accounts,
-			blocks,
+			blocksHeader,
```

Comment:
Thinking about further optimizations, maybe you could just pass only `this._db` into the functions, without `blocksHeader`? 
If I understand correctly, the `getBlockHeadersByIDs` function will basically iterate over all block headers to create the `blocksHeader` array and then you iterate over this array the second time. You could just read the DB stream in `getValidatorKeys` and initialize all validator keys in one database reading.

Not sure how clean this solution looks from the code quality perspective, but I think this could be an even better optimization.

Reply:
I applied the changes suggested here!